.github/workflows/deploy.yml

```
@@ -0,0 +1,32 @@
+name: Deploy built master on gh-pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  nikola_build:
+    runs-on: ubuntu-latest
+    name: 'Build pages'
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+    - name: Install dependencies
+      run: python -m pip install -r requirements.txt
+    - name: Build
+      run: nikola build
+    - name: Deploy
+      if: github.ref == 'refs/heads/master'
+      shell: bash
+      run: |
+          set -eou pipefail
+          git config --global user.name 'GithubAction'
+          git config --global user.email 'githubaction@invalid'
+          git pull --all
+          git remote add deploy https://${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY.git
```

A token shouldn't be needed within the same repo, not having to deal with authentication is one of the main advantages of using GH Actions right? Also, the secrets don't actually exist on this repo?

---

It was my understanding that `secrets.GITHUB_TOKEN` is a magic token that automatically exist for workflow running on pushed branches (and not pull-requests), but let's just try.
```suggestion
          git remote add deploy https://github.com/$GITHUB_REPOSITORY.git
```

---

> It was my understanding that `secrets.GITHUB_TOKEN` is a magic token that automatically exist for workflow running on pushed branches (and not pull-requests), but let's just try.

You were right, it works without.